src/component/handlers/edit/editOnCompositionStart.js

```diff
@@ -18,9 +18,10 @@ var EditorState = require('EditorState');
  * The user has begun using an IME input system. Switching to `composite` mode
  * allows handling composition input and disables other edit behavior.
  */
-function editOnCompositionStart(): void {
+function editOnCompositionStart(e): void {
   this.setRenderGuard();
   this.setMode('composite');
+  this._onCompositionStart(e);
```

Comment:
> Compositionstart can have a non-empty data event which indicate the part of text that is going to be replaced by composition. (on most major web browser).

We could check for `e.data` here, and if a value is present, use `Modifier.removeRange` to track the modified state and remove the range from the model. It would probably have to be flagged as natively rendered content to avoid re-rendering the editor during the composition. Would that help solve the issue here?

> One alternative solution would be to stop caring about those events and diff the DOM at strategical points, updating the model based on content and not on half broken events.

I'd prefer not to do DOM diffing.

Perhaps we could do something similar to the spellcheck handling in `editOnInput` -- grab the text content from the DOM and replace the equivalent range in the model. No diffing needed, just accommodate the DOM change.